src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -38,38 +57,88 @@ public PngEncoder() {
                 null);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code bufferedImage}.
+     * The new PngEncoder will use the provided {@code bufferedImage}.
+     *
+     * @param bufferedImage input image
+     * @return a new PngEncoder
+     */
     public PngEncoder withBufferedImage(BufferedImage bufferedImage) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code compressionLevel}.
+     * The new PngEncoder will use the provided {@code compressionLevel}.
+     *
+     * @param compressionLevel input image (must be between -1 and 9 inclusive)
+     * @return a new PngEncoder
+     */
     public PngEncoder withCompressionLevel(int compressionLevel) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code multiThreadedCompressionEnabled}.
+     * The new PngEncoder will use the provided {@code multiThreadedCompressionEnabled}.
+     *
+     * @param multiThreadedCompressionEnabled when {@code true}, multithreaded compression will be used
+     * @return a new PngEncoder
+     */
     public PngEncoder withMultiThreadedCompressionEnabled(boolean multiThreadedCompressionEnabled) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code srgbRenderingIntent}.
+     * The new PngEncoder will use the provided {@code srgbRenderingIntent}.
+     *
+     * @param srgbRenderingIntent todo
```

Comment:
Ww can find [the description of this in the PNG specification](http://www.libpng.org/pub/png/spec/1.1/PNG-Chunks.html#C.sRGB):
> If the sRGB chunk is present, the image samples conform to the sRGB color space [sRGB], and should be displayed using the specified rendering intent as defined by the International Color Consortium [ICC].

Perhaps we can write something like:
```
@param srgbRenderingIntent the rendering intent that should be used when displaying the image
```